src/handlers/invoice.rs

```diff
@@ -24,15 +32,22 @@ pub(crate) async fn invoice_add_upload(DatabaseConnection(mut conn): DatabaseCon
             vendor = match std::str::from_utf8(&data)? {
                 "metro" => Some(InvoiceVendor::Metro),
                 "bauhaus" => Some(InvoiceVendor::Bauhaus),
-                &_ => None
-            }
+                &_ => None,
+            };
+            parser = match vendor {
+                Some(InvoiceVendor::Metro) => Some(InvoiceParser::Regex(&METRO)),
+                Some(InvoiceVendor::Bauhaus) => Some(InvoiceParser::Regex(&BAUHAUS)),
+                None => None
+            };
```

Comment:
nit: Shouldn't this be part of the lib? It seems to be essential for its operation? Or is it here for easier extensibility?